torch/_dynamo/codegen.py

```diff
@@ -130,27 +133,24 @@ def __call__(self, value, allow_cache=True):
 
         if self.mutable_side_effects_from_source:
             # this is needed to get aliasing relationships right
-            # value.mutable_local.source will get mutated to hold `value`
+            # value.source will get mutated to hold `value`
             # mutable_side_effects_from_source=False is used to codegen the mutation
             # mutable_side_effects_from_source=True is used to codegen a reference
             from .side_effects import MutableSideEffects
 
             if isinstance(value.mutable_local, MutableSideEffects):
-                self(value.mutable_local.source)
+                self(value.source)
                 return
 
         if allow_cache:
-            if value.mutable_local and value.mutable_local in self.tempvars:
```

Comment:
This one feels like a separate change from this PR.  Though perhaps it is ok because VariableTrackers are now mutable.

Reply:
Yeah this is me trying to remove indexing on `MutableLocal`.